libcloudforensics/providers/gcp/internal/compute.py

```diff
@@ -963,6 +1004,110 @@ def Delete(
+    """
+    body = {}
+    if ip_addr is not None:
+      body['natIP'] = ip_addr
+    instances_client = self.GceApi().instances()
+    params = dict(
+      project=self.project_id,
+      zone=self.zone,
+      instance=self.name,
+      networkInterface=net_if,
+      body=body
+    )
+    try:
+      # Safe to unpack, as the response is not paged
+      response = common.ExecuteRequest(instances_client,
+                                       'addAccessConfig',
+                                       params)[0]
+    except HttpError as exception:
+      message = 'Unable to assign IP to {0:s}: {1:s}'.format(
+        self.name,
+        exception.error_details)
+      raise errors.ResourceCreationError(message, __name__)
+    self.BlockOperation(response, self.zone)
+
+  def RemoveExternalIps(self) -> Dict[str, str]:
+    """Removes any external IP of the instance, breaking ongoing connections.
+
+    Note that if the instance's IP address was static, that
+    the IP will still belong to the project.
+
+    Returns:
+      Dict[str, str]: A mapping from an instance's network
+        interfaces to the corresponding removed external IP.
+
+    Raises:
+      errors.ResourceDeletionError: If the removal did not succeed.
+    """
+    external_ip_addresses = {}
+    # Iterate through instance's network interfaces, removing
+    # all access configurations (NAT)
+    instance_info = self.GetOperation()
+    for network_interface in instance_info.get('networkInterfaces', []):
+      access_configs = network_interface.get('accessConfigs', [])
+      if len(access_configs) == 0:
+        # No way to access this network interface externally,
+        # skip the removal
+        continue
+      network_interface_name = network_interface['name']
+      # From the `get` operation response documentation, for
+      # the `networkInterfaces[].accessConfigs[]` field:
+      #
+      # > Currently, only one access config, ONE_TO_ONE_NAT, is
+      #   supported.
+      #
+      # It is thus safe to access only the first element.
+      access_config = access_configs[0]
+      access_config_name = access_config['name']
+      external_ip_address = access_config['natIP']
+      logger.info(
+        'Deleting access config for {0:s} (external IP: {1:s})'.format(
+          self.name,
+          external_ip_address,
+        ))
+      # Execute the IP address removal by deleting access config
+      gce_instance_client = self.GceApi().instances()
+      params = dict(
+        project=self.project_id,
+        zone=self.zone,
+        instance=self.name,
+        accessConfig=access_config_name,
+        networkInterface=network_interface_name)
+      try:
+        # Safe to unpack since this response is not paged
+        response = common.ExecuteRequest(gce_instance_client,
+                                         'deleteAccessConfig',
+                                         params)[0]
```

Comment:
Concerning these sort of non-paged calls, I'd rather write it as 
```suggestion
        # Safe to unpack since this response is not paged
        response, = common.ExecuteRequest(gce_instance_client,
                                         'deleteAccessConfig',
                                         params)
```
since as per the API reference I **really** don't expect to see more than one response here, and if it is the case things should not just continue working as usual. If we write it as suggested then we get at least get an unpacking error when we get more than one response.

This suggestion also entails adding `unbalanced-tuple-unpacking` to the disabled list in pylintrc.

Reply:
Imho there are a few problems with this approach: 
- It reduces readability (the `,` is not easy to spot, which makes the behaviour of this line hard to see). 
- Disabling `pylint` rules (especially at file level rather than line level) is generally speaking to be avoided, unless necessary. If you want to alert the user when the response contains more than one result, I'd suggest instead to log a warning:

```
responses = common.ExecuteRequest(gce_instance_client, 'deleteAccessConfig', params)
if len(responses) > 1:
  log.warning(....)
response = responses[0]  # Or access the member wherever used after in the code
```

API calls (their structures and responses) are bound to change over time. If a breaking change is introduced by the API provider, then our end-to-end tests are designed to catch them early.